assert.py

```diff
@@ -1,16 +1,97 @@
 import sys
 import os
 
-env_file = os.getenv('GITHUB_ENV')
-
-input_command_line = sys.argv[1]
-input_assert_file_path = sys.argv[2]
-input_specific_line = sys.argv[3]
-input_contains = sys.argv[4]
-input_expected_result = sys.argv[5]
-
-print(f"COMMAND LINE: {input_command_line}")
-print(f"ASSERT FILE PATH: {input_assert_file_path}")
-print(f"SPECIFIC LINE: {input_specific_line}")
-print(f"CONTAINS: {input_contains}")
-print(f"EXPECTED RESULT: {input_expected_result}")
+
+ENV_FILE = os.getenv('GITHUB_ENV')
+
+
+def main():
+	# This file should only run in the following conditions:
+	# It should receive exactly 6 arguments (including the file path).
+	if len(sys.argv) != 6:
```

Comment:
All the action inputs aren't required (mandatory) as you can check on the [action.yml file](https://github.com/GuillaumeFalourd/assert-command-line-output/blob/v3/action.yml) therefore I would remove this condition, e eventually just check the mandatory ones. What do you think?

Reply:
They still aren't required (by the workflow. It's just the script call that needs the workflow to pass in every argument). This is just to ensure all 6 sys.argv elements exist in the script. This required me to change the script call in the action.yml file though.

I ended up having to set this condition because the non-mandatory empty inputs turn into empty strings when we pass them in to "python assert.py <... ... ... ... ...>" and these empty strings are ignored by sys.argv. That ends up turning into a list index out of range error when accessing sys.argv the way the script currently is. It also becomes a bit troublesome (though not impossible) to figure out what options user chose, as what's at each index will change depending on which option the user left empty.

I figured it would be a lot easier to code up and straightforward for readers to understand if we enforce the existence of those outputs by adding any character (such as "_") before the users input in the script call (and later removing that character once the script is run). That way, we still get the same user input, what was left empty in the workflow call will be empty strings in the script, but this way we will be sure that every argument exist (at least as empty strings) and are always in the same position.

Someone using this action would still use it normally. It's the action itself that adds the "_" to the non-mandatory inputs (ensuring all 5 inputs exist) and the script removes them so we still get the actual user input.